src/libmeasurement_kit/ooni/utils.cpp

```diff
@@ -139,13 +139,25 @@ ErrorOr<std::string> GeoipDatabase::resolve_asn(std::string ip,
 }
 
 std::string extract_html_title(std::string body) {
-  std::regex TITLE_REGEXP("<title>([\\s\\S]*?)</title>", std::regex::icase);
-  std::smatch match;
-
-  if (std::regex_search(body, match, TITLE_REGEXP) && match.size() > 1) {
-    return match.str(1);
+  // +-+-+---------------+-+-----+
+  // |f|o|...............|r| '\0'|
+  // +-+-+---------------+-+-----+
+  //  ^                       ^
+  // body.data()         body.data() + body.size()
+  auto zerop = body.data() + body.size();
+  auto begin_title = strcasestr(body.data(), "<title>");
+  if (begin_title == nullptr) {
+    return "";
+  }
+  begin_title += sizeof "<title>";
+  assert(begin_title <= zerop); // Must always hold
+  auto end_title = strcasestr(begin_title, "</title>");
+  if (end_title == nullptr) {
+    return "";
   }
-  return "";
+  assert(end_title >= begin_title); // Should really not happen
+  auto siz = static_cast<size_t>(end_title - begin_title);
+  return std::string{begin_title, siz};
```

Comment:
IMHO the code serves the goal to get rid of regex engine, but...
- I'd rather ship "known good" regex engine if there are other regexes in the MK (maybe there are none of them, I don't know)
- Seems, `\0` breaks strcasestr()
- Wild-life example that breaks this code is `<title id="pageTitle">Facebook</title>` from view-source:https://www.facebook.com/ :-)